File: AccountManagement/Forms/NewContactForm.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;

using EnumLib;
using PROPMANS.BL.AccountManagement;


namespace PROPMANS.VIEW.AccountManagement
{
    public partial class NewContactForm : ComponentFactory.Krypton.Toolkit.KryptonForm,
        INewContact
    {
        NewContactPresenter presenter;
        #region "Form Instance"
        private static NewContactForm aForm;
        public static NewContactForm Instance()
        {
            if (aForm == null)
            {
                aForm = new NewContactForm();
            }
            return aForm;
        }

        private void Form_Disposed(object sender, System.EventArgs e)
        {
            aForm = null;
        }
        #endregion
        public NewContactForm()
        {
            InitializeComponent();
            this.Disposed += Form_Disposed;
            presenter = new NewContactPresenter(this);
        }

        #region PROPERTIES
        private string _customerID;
        public string CustomerID
        {
            get { return _customerID; }
            set { _customerID = value; }
        }
        #endregion

        #region INTERFACE
        private CustomerContact _customerContact;
        public CustomerContact CustomerContact
        {
            get
            {                
                if (_customerContact == null) _customerContact = new CustomerContact();

                _customerContact.ContactType = (ContactType)cboContactType.SelectedValue;
                _customerContact.ContactLocation = (ContactLocation)cboContactLocation.SelectedValue;
                _customerContact.Details = txtContact.Text.Trim();                
                if (RbContactYes.Checked)
                {
                    _customerContact.Active = ActiveStatus.Yes;
                }
                else { _customerContact.Active = ActiveStatus.No; }
                if (chkPreferred.Checked) {
                    _customerContact.Preffered = ActiveStatus.Yes;
                }
                else { _customerContact.Preffered = ActiveStatus.No; }
                return _customerContact;
            }
            set
            {
                if (value != null)
                {
                    cboContactType.SelectedValue = (Int32)value.ContactType;
                    cboContactLocation.SelectedValue = (Int32)value.ContactLocation;
                    txtContact.Text = value.Details;
                    if (value.Active.Equals("Yes"))
                    {
                        RbContactYes.Checked = true;
                    }
                    else { RbContactNo.Checked = false; }
                    if (value.Preffered.Equals("Yes")) {
                        chkPreferred.Checked = true;
                    }
                    else { chkPreferred.Checked = false; }
                    _customerID = value.Customer.CustomerID;
                    _customerContact = value;
                }                
            }
        }

        public void LoadContactType(List<EnumItem> src)
        {
            cboContactType.DisplayMember = "Name";
            cboContactType.ValueMember = "Value";
            cboContactType.DataSource = src;
        }

        public void LoadContactLocation(List<EnumItem> src)
        {
            cboContactLocation.DisplayMember = "Name";
            cboContactLocation.ValueMember = "Value";
            cboContactLocation.DataSource = src;
        }

        public void DisplaySaveStatus(bool status)
        {
            if (status)
            {
                MessageBox.Show("New Contact Successfully Added!");
                this.DialogResult = System.Windows.Forms.DialogResult.OK;
                this.Close();
            }
            else
            {
                MessageBox.Show("Failed to add new Contact!");
            }
        }

        public void DisplayUpdateStatus(bool status)
        {
            if (status)
            {
                MessageBox.Show("Contact Successfully Updated!");
                this.DialogResult = System.Windows.Forms.DialogResult.OK;
                this.Close();
            }
            else
            {
                MessageBox.Show("Failed to Update Contact!");
            }
        }
        #endregion
        
        #region EVENTS
        private void btnCancel_Click(object sender, EventArgs e)
        {
            this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.Close();
        }
        private void btnSave_Click(object sender, EventArgs e)
        {
            if (this.btnSave.Text == "&SAVE")
            {
                if (MessageBox.Show("ADD NEW CONTACT?", "CONFIRM SAVING",
                MessageBoxButtons.YesNo, MessageBoxIcon.Question, MessageBoxDefaultButton.Button2) == DialogResult.Yes)
                {
                    if (IsInputValid())
                    {
                        presenter.SaveContact(_customerID);
                        //presenter.SaveContact();
                    }
                }
            }
            else
            {
                if (MessageBox.Show("UPDATE CONTACT?", "CONFIRM SAVING",
                MessageBoxButtons.YesNo, MessageBoxIcon.Question, MessageBoxDefaultButton.Button2) == DialogResult.Yes)
                {
                    if (IsInputValid())
                    {
                        presenter.UpdateContact();
                    }
                }
            }
            
        }
        #endregion

        #region PRIVATE FUNCTION(S)
        private bool IsInputValid()
        {
            bool result = true;
            errMsg.Clear();
            errMsg.SetError(txtContact, "");
            if (String.IsNullOrWhiteSpace(txtContact.Text))
            {
                errMsg.SetError(txtContact, "Invalid Contact Details!");
                result = false;
            }

            return result;
        }
        #endregion



        
    }
}

File: PROPMANS.ACCOUNT_MGMT/Customer Accounts/NewContactForm.vb
Imports ComponentFactory.Krypton.Toolkit
Imports BCL


Imports BCL.CustomerAccounts
Imports PROPMANS.BASE_MOD

Public Class NewContact

#Region "Properties"

    Public _customerId As String

    Public Property CustomerId() As String
        Get
            Return _customerId
        End Get
        Set(ByVal Value As String)
            _customerId = Value
        End Set
    End Property

    Public _contactId As String

    Public Property ContactId() As String
        Get
            Return _contactId
        End Get
        Set(ByVal Value As String)
            _contactId = Value
        End Set
    End Property

    Private _contactLocation As ContactLocation
    Public Property ContactLocation() As ContactLocation
        Get
            Return _contactLocation
        End Get
        Set(ByVal value As ContactLocation)
            _contactLocation = value
        End Set
    End Property

    Private _contactType As ContactType
    Public Property ContactType() As ContactType
        Get
            Return _contactType
        End Get
        Set(ByVal value As ContactType)
            _contactType = value
        End Set
    End Property

#End Region

#Region "Form & Control Events"

    Private Sub NewContact_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

        UISetup()

    End Sub

    Private Sub btnSave_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSave.Click

        If txtContact.Text = String.Empty Then

            MessageBox.Show("Please Input Contact", "Value Required", MessageBoxButtons.OK, MessageBoxIcon.Information)

        Else
            If btnSave.Text = "&UPDATE" Then
                UpdateRecord()
            Else
                InsertRecord()
            End If

        End If

    End Sub

    Private Sub btnCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancel.Click

        Me.DialogResult = Windows.Forms.DialogResult.Cancel
        Me.Close()

    End Sub
#End Region

#Region "Local Procedures"

    Private Sub UISetup()

    End Sub

    Private Sub InsertRecord()

        Dim InsertContact As New CustomerContact

        With InsertContact

            .CustomerId = Me.CustomerId
            .ContactType = ComboBoxContactType.SelectedValue
            .ContactLoc = ComboBoxContactSubType.SelectedValue
            .Details = txtContact.Text

            If RbContactYes.Checked Then
                .Active = Active.Yes
            Else
                .Active = Active.No
            End If


        End With
        CustomerContact.InsertContact(InsertContact)

        Me.DialogResult = Windows.Forms.DialogResult.OK

        Me.Close()

    End Sub

    Private Sub UpdateRecord()

        Dim UpdateContact As New CustomerContact

        With UpdateContact

            .CustomerId = Me.CustomerId
            .ContactId = Me.ContactId
            .ContactType = ComboBoxContactSubType.SelectedValue
            .ContactLoc = ComboBoxContactType.SelectedValue
            .Details = txtContact.Text

            If RbContactYes.Checked Then

                .Active = Active.Yes

            Else

                .Active = Active.No

            End If

        End With

        CustomerContact.UpdateContact(UpdateContact)

        Me.DialogResult = Windows.Forms.DialogResult.OK

        Me.Close()

    End Sub

    Private Sub DisplayRecord()

        Common.BindComboBoxtoList(ComboBoxContactType, GetType(ContactType))
        Common.BindComboBoxtoList(ComboBoxContactSubType, GetType(ContactLocation))

    End Sub

    Public Sub ViewRecord()

        Common.BindComboBoxtoList(ComboBoxContactType, GetType(ContactType))
        ComboBoxContactType.SelectedValue = ContactType

        Common.BindComboBoxtoList(ComboBoxContactSubType, GetType(ContactLocation))
        ComboBoxContactSubType.SelectedValue = ContactLocation

    End Sub

#End Region

End Class
